### ISSUE TITLE
Cover resizing is incorrect

### ISSUE THREAD

**Issue by btamas86** [Intervention/image#1359]
**Describe the bug**
I use the "cover" type of resizing and I noticed that if the original image is odd in size, the resizing is incorrect.

Original image size: 375x250
Requested size: 240x90
Received size: 239x90

So it is rounded down, not up, which I think is a problem.

**Code Example**
```php
$imageContent = file_get_contents('image.jpg');
$image = (new Intervention\Image\ImageManager(new Intervention\Image\Drivers\Gd\Driver))->read($imageContent);
$image->cover(240, 90, 'center');
$imgEncoded = $image->encodeByMediaType(quality: 85);

response((string)$imgEncoded)
    ->withHeaders([
        'Content-type' => $imgEncoded->mediaType(),
    ]);
```

**Images**
Original:
![4k-wallpaper-background-beautiful-853199-horizontal](https://github.com/Intervention/image/assets/4123828/ffebe529-f3b1-4280-a7e3-019406dc6cb9)
Resized:
<img width="482" alt="image" src="https://github.com/Intervention/image/assets/4123828/edd49ade-2266-4ede-9372-5cbc15215bb4">

**Environment (please complete the following information):**
 - PHP Version: 8.2.17 (php:8.2.17-apache)
 - OS: Debian GNU/Linux 12 (bookworm), 6.6.26-linuxkit
 - Intervention Image Version: 3.6.4
 - GD or Imagick: GD

**Comment by olivervogel** [comment]
Thanks for reporting. Yes, I can confirm the bug. Unfortunately it happens with GD and Imagick. I will look into it.

**Comment by btamas86** [comment]
Thank you very much!

**Comment by olivervogel** [comment]
This is now fixed in `3.6.5`.
